src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -96,13 +137,415 @@ void AreaTrigger::Update(uint32 p_time)
         Remove(); // expired
 
     WorldObject::Update(p_time);
+    _timeSinceCreated += p_time;
+
+    if (GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_ATTACHED))
+    {
+        if (Unit* target = GetTarget())
+            GetMap()->AreaTriggerRelocation(this, target->GetPositionX(), target->GetPositionY(), target->GetPositionZ(), target->GetOrientation());
```

Comment:
What if target changes map? it can also happen ([same spell again](http://www.wowhead.com/spell=157411/windwalking))